File: data/gpt-4o/readable_format_HPEC/cpp/file225.cpp
void kernel_lu(int n, float a[n][n]) 
{
  int i, j, k;
  
  for (k = 0; k < n; k++)
  {
    for (j = k + 1; j < n; j++)
    {
      a[j][k] = a[j][k] / a[k][k];
    }
    for (i = k + 1; i < n; i++)
    {
      for (j = k + 1; j < n; j++)
      {
        a[j][i] = a[j][i] - (a[k][i] * a[j][k]);
      }
    }
  }
}

File: data/CodeLlama/readable_format_HPEC/fortran/file225.f90
subroutine kernel_lu(n, a)
implicit none

DATA_TYPE, dimension(n, n) :: a
integer :: n
integer :: i, j, k

!$pragma scop
do k = 1, _PB_N
do j = k + 1, _PB_N
a(j, k) = a(j, k) / a(k, k)
end do
do i = k + 1, _PB_N
do j = k + 1, _PB_N
a(j, i) = a(j, i) - (a(k, i) * a(j, k))
end do
end do
end do
!$pragma endscop
end subroutine
